admin/src/app/Components/hotel-for-admin/hotel-for-admin.component.ts

```diff
@@ -48,11 +54,11 @@ export class HotelForAdminComponent implements OnInit {
   }
 
   public editHotel(): void {
-    if (this.hotelId === undefined) {
+    if (this.hotel === undefined) {
       return;
     }
     this.hotelService
-      .editHotel(this.hotelId, this.changeHotelForm.value as Hotel)
+      .editHotel(this.hotel.id, this.hotel)
```

Comment:
You are already using all the information about the hotel as your second argument. Why do you need to pass the id separately?